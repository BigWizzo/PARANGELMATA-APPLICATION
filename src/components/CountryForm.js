import React, { useState } from 'react';

const CountryForm = ({ onSub }) => {
  const [nat, setNat] = useState('');
  const onSubmit = (e) => {
    e.preventDefault();
    if (nat) {
      setNat(nat.charAt(0).toUpperCase() + nat.slice(1));
    }

    if (!nat) {
      alert('Please enter Country Name');
    }

    onSub({ nat });
  };

  return (
    <>
      <form className="country-form" onSubmit={onSubmit}>
        <div className="my-form mb-3">
          <input
            className="form-control"
            type="text"
            value={nat}
            placeholder="Enter Country"
            onChange={(e) => setNat(e.target.value)}
          />
          <div id="emailHelp" className="form-text">
            We'll never share your email with anyone else.
          </div>
        </div>
        <div className="d-grid gap-2 d-md-block">
          <button
            type="submit"
            className="btn btn-primary btn-dark"
            value="Submit"
          >
            Submit
          </button>
        </div>
      </form>
    </>
  );
};

export default CountryForm;
