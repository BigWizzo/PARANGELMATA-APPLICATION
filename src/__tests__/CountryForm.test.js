import React from 'react';
import CountryForm from '../components/CountryForm';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store, persistor } from '../redux/store';
import { PersistGate } from 'redux-persist/integration/react';

test('should render the submit form to the App DOM', () => {
  const component = renderer.create(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <CountryForm />
      </PersistGate>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
