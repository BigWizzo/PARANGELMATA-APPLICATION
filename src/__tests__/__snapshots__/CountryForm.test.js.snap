// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the submit form to the App DOM 1`] = `
<form
  className="country-form m-auto mb-5"
  onSubmit={[Function]}
>
  <div
    className="my-form mb-3"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Enter Country"
      type="text"
      value=""
    />
  </div>
  <div
    className="d-grid gap-2 d-block"
  >
    <button
      className="btn btn-primary font-weight-bolder"
      type="submit"
      value="Submit"
    >
      Submit
    </button>
  </div>
</form>
`;
